design/kanister-kopia-integration.md

```diff
@@ -0,0 +1,471 @@
+   azure_storage_key: <redacted>
+   # optional: base64 encoded value for the storage enevironment.
+   # Acceptable values are AzureCloud, AzureChinaCloud, AzureUSGovernment, AzureGermanCloud
+   azure_storage_environment: <redacted>
+```
+
+The password used while creating the Kopia repository is provided by the user in the `Secret` resource referenced by `spec.repository.passwordSecret`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+   name: repository-monitoring-password
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: Opaque
+data:
+   repo-password: <redacted>
+```
+
+The server admin credentials, and TLS sensitive data are stored in the `Secret` resources referenced by the `spec.server.adminSecret` and
+`spec.server.tlsSecret` properties. For example,
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+   name: repository-monitoring-server-admin
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: Opaque
+data:
+   username: <redacted>
+   password: <redacted>
+---
+apiVersion: v1
+kind: Secret
+metadata:
+   name: repository-monitoring-server-tls
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: kubernetes.io/tls
+data:
+   tls.crt: |
+      <redacted>
+   tls.key: |
+      <redacted>
+```
+
+The `spec.server.accessSecret` property provides a list of access credentials used by data mover clients to authenticate
+with the Kopia repository server. This is discussed in more detail in the
+[Server Access Users Management](#server-access-users-management) section.
+
+The Kopia repository server `Pod` resource can be accessed through a K8s `Service`. The `spec.server.labelSelector` property
+is used to determine the labels used in the `podSelector` of the `NetworkPolicy` resource that controls the ingress
+traffic to the repository server.
+
+The `status` subresource provides conditions and status information to ensure that the controller does not attempt to re-create
+the repository server during restart.
+
+#### Repository Server Lifecycle
+
+When a `RepositoryServer` resource is created, the `RepositoryController` controller responds by creating a `Pod`, `Service`,
+and a `NetworkPolicy` as mentioned above.
+These resources are cleaned up when the `RepositoryServer` resource is deleted.
+
+#### Server-side Setup
+
+Once the pod is running, the controller executes a set of Kopia CLI commands as follows.
+
+> 📝 The examples in this section use S3 for illustration purposes.
+
+- Establish a connection to the Kopia repository
+
+```sh
+kopia repository connect s3 \
+   --bucket=my-bucket \
+   --access-key=<redacted> \
+   --secret-access-key=<redacted> \
+   [--endpoint=https://foo.example.com \]
+   [--prefix=my-prefix \]
+   [--region=us-west-1 \]
+   --password=<redacted> \
+   --override-hostname=<hostname> \
+   --override-username=<username>
```

Comment:
Do we know if this is still necessary or is `kopia server start` smart enough to figure it out if we add all the repo info into the `repo.config` file?

Reply:
A connect operation is required even if the configuration is added to a file.
We are exploring ways to utilize the `kopia repository connect from-config` option provided by Kopia to supply all the information via a single config file.